### ISSUE TITLE
"exceeding bitshifts" lint does not always work in debug mode on playground

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#70660]
I tried this code:

```rust
#![crate_type = "lib"]
pub fn foo(x: i32) -> i32 {
    x << 33
}
```
I am building in debug mode on [the playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f4aeb4486b7625ef37fe9e2d4bc25ca8).

I expected to see this happen: it should warn about the exceeding bitshift (like it does in release mode).

Instead, this happened: there is no warning,

### Meta

Possibly related to https://github.com/rust-lang/rust/issues/69021

**Comment by RalfJung** [comment]
I am quite puzzled why this does not get caught by [this test](https://github.com/rust-lang/rust/blob/99009bf31992fb9257b84550dbf7cb509f987952/src/test/ui/lint/lint-exceeding-bitshifts.rs) which I added specifically to catch cases like this. Isn't the "noopt" revision equivalent to debug mode?
```
// revisions: noopt opt opt_with_overflow_checks
//[noopt]compile-flags: -C opt-level=0
//[opt]compile-flags: -O
//[opt_with_overflow_checks]compile-flags: -C overflow-checks=on -O
```

**Comment by RalfJung** [comment]
Oh, well that is strange... when I locally just run `rustc` on the same code (with no flags), I get the expected warning.

@shepmaster what is the exact invocation that playground uses with "Debug" mode to build the code?

**Comment by RalfJung** [comment]
Oh lol, I only checked stable Rust; this got fixed by one of my PRs from February and those are not on stable yet... d'oh.

**Comment by RalfJung** [comment]
Also, we have to make sure that there always is some ICE we can trigger in rustc, because that's a good way to get the actual compiler flags. ;)

**Comment by shepmaster** [comment]
> there always is some ICE we can trigger in rustc

Isn’t that what `break rust` is for?

**Comment by RalfJung** [comment]
@shepmaster yeah but see https://github.com/rust-lang/rust/issues/70661
